packages/cdktf-cli/lib/watch.ts

```diff
@@ -0,0 +1,173 @@
+import path from "path";
+import {
+  CdktfProject,
+  CdktfProjectOptions,
+  ExecutionOptions,
+} from "./cdktf-project";
+import { Errors } from "./errors";
+import * as fs from "fs";
+import * as chokidar from "chokidar";
+import { logger } from "./logging";
+import { CdktfStack } from "./cdktf-stack";
+
+// In this very first iteration we will find out which files to watch by asking the user to provide the files
+// We default to something reasonable per language and write it into the cdkt.json file
+// Mid-Term we might want to add a WatchFile / WatchDir construct that we can use (e.g. in assets) and that a user can use to specify their watch behaviour
```

Comment:
Do we have follow-up issues for that?

Reply:
Created https://github.com/hashicorp/terraform-cdk/issues/1668 for this